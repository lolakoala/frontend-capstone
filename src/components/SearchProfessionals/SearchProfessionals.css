.search-options {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  margin-top: 30px;
}
