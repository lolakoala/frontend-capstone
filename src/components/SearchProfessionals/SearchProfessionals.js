import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { DropDownMenu, MenuItem } from 'material-ui';

class SearchProfessionals extends Component {
  constructor() {
    super();
    this.state = {
      searchQuery: 'insurance',
      searchTopic: ''
    };
  }

  componentWillMount() {
    this.props.getInsuranceList();
    this.props.getSpecialtyList()
  }

  handleQuery = (event, index, value) => this.setState({ searchQuery: value });

  render() {
    const insuranceList

    return (
      <div>
        <DropDownMenu value={this.state.searchQuery} onChange={this.handleQuery}>
          <MenuItem value="select query" primaryText="Select search query." />
          <MenuItem value="insurance" primaryText="Search by insurance." />
          <MenuItem value="specialty" primaryText="Search by specialty." />
        </DropDownMenu>
        {this.state.value === 'insurance' ? insuranceList : null}
        {this.state.value === 'specialty' ? specialtyList : null}
        {this.state.searchTopic ? submitButton : null}
      </div>
    );
  }
}

export default SearchProfessionals;

SearchProfessionals.propTypes = {
  getInsuranceList: PropTypes.func,
  getSpecialtyList: PropTypes.func
};
