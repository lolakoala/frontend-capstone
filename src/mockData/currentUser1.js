const currentUser1 = {
  email: 'lola@gmail.com',
  city: 'Denver',
  userName: 'lolakoala',
  aboutMe: 'Lola loves doggos.'
};

export default currentUser1;
