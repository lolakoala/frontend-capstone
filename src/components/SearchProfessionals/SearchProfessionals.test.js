import React from "react";
import SearchProfessionals from './SearchProfessionals.js';
import { shallow, configure, mount } from "enzyme";
import currentUser1 from '../../mockData/currentUser1.js';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('SearchProfessionals', () => {
  const mockFn = jest.fn();
  const allChallenges = ['depression', 'anxiety'];
  const wrapper = shallow(<SearchProfessionals
    getInsuranceList={mockFn}
    getSpecialtyList={mockFn}
    insuranceList={['aetna']}
    specialtyList={[]}
    search={mockFn}
    currentUser={currentUser1}
  />);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match snap', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call 2 functions on mount', () => {
    expect(mockFn).toHaveBeenCalledTimes(2);
  });

  it('should have drop down menu', () => {
    const ddm = wrapper.find('DropDownMenu');

    expect(ddm.length).toEqual(1);
  });

  // cannot test further b/c cannot figure out how to give component
  // muiTheme context from test when using mount instead of shallow
  it.skip('should change state on drop down change', () => {
    const ddm = wrapper.find('DropDownMenu');

    ddm.simulate('change', { target: { value: 'insurance' } });
    expect(wrapper.state().searchQuery).toEqual('insurance');
  });


  // it('should have one button after state change', () => {
  //   const button = wrapper.find('RaisedButton');
  //
  //   expect(button.length).toEqual(1);
  // });

  // tests below here are copied from SearchBuddies and need to be modified
  //
  // // this test returns undefined... not sure why
  // it.skip('should change state on change of drop down menu', () => {
  //   const ddm = wrapper.find('DropDownMenu');
  //
  //   ddm.simulate('change', { target: { value: allChallenges[1] } });
  //   expect(wrapper.state().value).toEqual(allChallenges[1]);
  // });
  //
  // it('should call function when button clicked', () => {
  //   const button = wrapper.find('RaisedButton');
  //
  //   button.simulate('click');
  //   expect(mockFn).toHaveBeenCalledTimes(2);
  // });
});
