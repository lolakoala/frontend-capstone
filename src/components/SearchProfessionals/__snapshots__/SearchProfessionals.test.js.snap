// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchProfessionals should match snap 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchProfessionals
    currentUser={
        Object {
            "aboutMe": "Lola loves doggos.",
            "city": "Denver",
            "email": "lola@gmail.com",
            "userName": "lolakoala",
          }
    }
    getInsuranceList={[Function]}
    getSpecialtyList={[Function]}
    insuranceList={
        Array [
            "aetna",
          ]
    }
    search={[Function]}
    specialtyList={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <DropDownMenu
          anchorOrigin={
                    Object {
                              "horizontal": "left",
                              "vertical": "top",
                            }
          }
          animated={true}
          autoWidth={true}
          disabled={false}
          iconButton={<NavigationArrowDropDown />}
          maxHeight={500}
          multiple={false}
          onChange={[Function]}
          openImmediately={false}
          value="select query"
>
          <MenuItem
                    anchorOrigin={
                              Object {
                                        "horizontal": "right",
                                        "vertical": "top",
                                      }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="Select search query."
                    targetOrigin={
                              Object {
                                        "horizontal": "left",
                                        "vertical": "top",
                                      }
                    }
                    value="select query"
          />
          <MenuItem
                    anchorOrigin={
                              Object {
                                        "horizontal": "right",
                                        "vertical": "top",
                                      }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="Search by insurance."
                    targetOrigin={
                              Object {
                                        "horizontal": "left",
                                        "vertical": "top",
                                      }
                    }
                    value="insurance"
          />
          <MenuItem
                    anchorOrigin={
                              Object {
                                        "horizontal": "right",
                                        "vertical": "top",
                                      }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="Search by specialty."
                    targetOrigin={
                              Object {
                                        "horizontal": "left",
                                        "vertical": "top",
                                      }
                    }
                    value="specialty"
          />
</DropDownMenu>,
        null,
        null,
        null,
      ],
      "className": "search-options",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "anchorOrigin": Object {
            "horizontal": "left",
            "vertical": "top",
          },
          "animated": true,
          "autoWidth": true,
          "children": Array [
            <MenuItem
              anchorOrigin={
                            Object {
                                          "horizontal": "right",
                                          "vertical": "top",
                                        }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="Select search query."
              targetOrigin={
                            Object {
                                          "horizontal": "left",
                                          "vertical": "top",
                                        }
              }
              value="select query"
/>,
            <MenuItem
              anchorOrigin={
                            Object {
                                          "horizontal": "right",
                                          "vertical": "top",
                                        }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="Search by insurance."
              targetOrigin={
                            Object {
                                          "horizontal": "left",
                                          "vertical": "top",
                                        }
              }
              value="insurance"
/>,
            <MenuItem
              anchorOrigin={
                            Object {
                                          "horizontal": "right",
                                          "vertical": "top",
                                        }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="Search by specialty."
              targetOrigin={
                            Object {
                                          "horizontal": "left",
                                          "vertical": "top",
                                        }
              }
              value="specialty"
/>,
          ],
          "disabled": false,
          "iconButton": <NavigationArrowDropDown />,
          "maxHeight": 500,
          "multiple": false,
          "onChange": [Function],
          "openImmediately": false,
          "value": "select query",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "Select search query.",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "select query",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "Search by insurance.",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "insurance",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "Search by specialty.",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "specialty",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      null,
      null,
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DropDownMenu
            anchorOrigin={
                        Object {
                                    "horizontal": "left",
                                    "vertical": "top",
                                  }
            }
            animated={true}
            autoWidth={true}
            disabled={false}
            iconButton={<NavigationArrowDropDown />}
            maxHeight={500}
            multiple={false}
            onChange={[Function]}
            openImmediately={false}
            value="select query"
>
            <MenuItem
                        anchorOrigin={
                                    Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                        }
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        primaryText="Select search query."
                        targetOrigin={
                                    Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                        }
                        value="select query"
            />
            <MenuItem
                        anchorOrigin={
                                    Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                        }
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        primaryText="Search by insurance."
                        targetOrigin={
                                    Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                        }
                        value="insurance"
            />
            <MenuItem
                        anchorOrigin={
                                    Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                        }
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        primaryText="Search by specialty."
                        targetOrigin={
                                    Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                        }
                        value="specialty"
            />
</DropDownMenu>,
          null,
          null,
          null,
        ],
        "className": "search-options",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "anchorOrigin": Object {
              "horizontal": "left",
              "vertical": "top",
            },
            "animated": true,
            "autoWidth": true,
            "children": Array [
              <MenuItem
                anchorOrigin={
                                Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="Select search query."
                targetOrigin={
                                Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                }
                value="select query"
/>,
              <MenuItem
                anchorOrigin={
                                Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="Search by insurance."
                targetOrigin={
                                Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                }
                value="insurance"
/>,
              <MenuItem
                anchorOrigin={
                                Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="Search by specialty."
                targetOrigin={
                                Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                }
                value="specialty"
/>,
            ],
            "disabled": false,
            "iconButton": <NavigationArrowDropDown />,
            "maxHeight": 500,
            "multiple": false,
            "onChange": [Function],
            "openImmediately": false,
            "value": "select query",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "primaryText": "Select search query.",
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
                "value": "select query",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "primaryText": "Search by insurance.",
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
                "value": "insurance",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "primaryText": "Search by specialty.",
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
                "value": "specialty",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        null,
        null,
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
